### DISCUSSION TITLE
How do I format/transform XML file that I import into the code snippet?

### DISCUSSION THREAD

**Discussion by akil-rails** [vuejs/vitepress#1125] [Q&A]
I'am attempting to document a pretty large XML file. The way I am going about is to document parts of the XML. 

I am currently having to split the large XML files into multiple smaller ones (format them), and import them as described in the [docs](https://vitepress.vuejs.org/guide/markdown.html#import-code-snippets) . 

Is there a way, I could keep one XML file, and do the rest in markdown/via a script?

**Comment by brc-dd** [comment]
> Is there a way, I could keep one XML file, and do the rest in markdown/via a script?

Define and use regions? Its mentioned on that docs section near the end. Not sure if they work with XML though.

**Reply by akil-rails** [reply]
Regions work with XML; but they aren't sufficient - i also need to reformat the xml fragment..

**Comment by akil-rails** [comment]
For code blocks, is there a way to provide the data as a prop in the .md file  - such as follows ? 



```
``
{{ data }}
``

<script setup>
import { ref} from 'vue'
import dt from './my.xml?raw'

const data = ref(dt)
</script>
```

**Reply by brc-dd** [reply]
Yes, it will be in the next release ([docs](https://vitepress.vuejs.org/guide/using-vue.html#escaping)). You'll be able to use `-vue` prefix in code blocks' language to enable Vue-style interpolations. For example:

````md
```xml-vue
{{ data }}
```
````
